src/Symfony/Component/Security/Core/Security.php

```diff
@@ -11,15 +11,63 @@
 
 namespace Symfony\Component\Security\Core;
 
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
 /**
- * This class holds security information.
- *
- * @author Johannes M. Schmitt <schmittjoh@gmail.com>
+ * Helper class for commonly-needed security tasks.
  */
 final class Security
 {
     const ACCESS_DENIED_ERROR = '_security.403_error';
     const AUTHENTICATION_ERROR = '_security.last_error';
     const LAST_USERNAME = '_security.last_username';
     const MAX_USERNAME_LENGTH = 4096;
+
+    private $container;
+
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * @return UserInterface|null
+     */
+    public function getUser()
+    {
+        if (!$token = $this->getToken()) {
+            return null;
+        }
+
+        $user = $token->getUser();
+        if (!is_object($user)) {
```

Comment:
Can't the user be a simple string in some cases?
I mean, the `TokenInterface` says this:

```php
/**
 * Returns a user representation.
 *
 * @return mixed Can be a UserInterface instance, an object implementing a __toString method,
 *               or the username as a regular string
 *
 * @see AbstractToken::setUser()
 */
public function getUser();
```

So instead, this should return `mixed` instead of `UserInterface`...

Reply:
Yep, but I'm purposely not allowing for people who use a string User to make use of this feature. We do that in other places. Allowing it to be a string makes everyone's user experience worse. So, you *can* do this, but you need to use the lower-level tools :)